### PR TITLE
Checkmarx parser: fix empty filename

### PR DESCRIPTION
**Pull request by damiencarol**
- [x] fix a bug when file name is empty #5636
- [x] improve language stats data (number of files+fix stats in the UI)
- [x] remove some useless fake data (mitigation, impact and references blank)

### CODE REVIEW

File: dojo/tools/checkmarx/parser.py [hunk 1]
```diff
@@ -45,21 +42,17 @@ def get_description_for_scan_types(self, scan_type):
...
 
-    # FIXME get rid of local variables
-    language_list = []
-    mitigation = 'N/A'
-    impact = 'N/A'
-    references = ''
-
     def get_findings(self, filename, test):
         cxscan = ElementTree.parse(filename)
+        # FIXME get rid of local variable
```
Comment (StefanFl) [inline dojo/tools/checkmarx/parser.py:47]: Will you fix this within this PR?

File: dojo/tools/checkmarx/parser.py [hunk 2]
```diff
@@ -79,8 +72,10 @@ def get_findings(self, filename, test):
 
                 if language is not None:
                     findingdetail = "{}**Language:** {}\n".format(findingdetail, language)
-                    if language not in self.language_list:
-                        self.language_list.append(language)
+                    if language not in language_list:
+                        language_list[language] = 1
+                    else:
+                        language_list[language] = language_list[language] + 1
```
Comment (StefanFl) [inline dojo/tools/checkmarx/parser.py:78]: How much sense makes it to count the number of files for a language? You only get the number of files with vulnerabilities, right?

File: dojo/tools/checkmarx/parser.py [hunk 3]
```diff
@@ -110,7 +105,10 @@ def process_result_file_name_aggregated(self, dupes, findingdetail, query, resul
         titleStart = query.get('name').replace('_', ' ')
         description, lastPathnode = self.get_description_file_name_aggregated(query, result)
         sinkFilename = lastPathnode.find('FileName').text
-        title = "{} ({})".format(titleStart, ntpath.basename(sinkFilename))
+        if sinkFilename:
+            title = "{} ({})".format(titleStart, sinkFilename.split("/")[-1])
```
Comment (StefanFl) [inline dojo/tools/checkmarx/parser.py:109]: What does happen, when a filename doesn't have a `/`? And does this work when the check is done on Windows (if this is possible)?

File: dojo/tools/checkmarx/parser.py [hunk 4]
```diff
@@ -210,7 +203,7 @@ def process_result_detailed(self, dupes, findingdetail, query, result, find_date
             # pathId is the unique id from tool which means that there is basically no aggregation except real duplicates
             aggregateKeys = "{}{}{}{}{}".format(categories, cwe, name, sinkFilename, pathId)
             if title and sinkFilename:
-                title = "{} ({})".format(title, ntpath.basename(sinkFilename))
+                title = "{} ({})".format(title, sinkFilename.split("/")[-1])
```
Comment (StefanFl) [inline dojo/tools/checkmarx/parser.py:206]: see my question on line 109

**Review by StefanFl — COMMENTED** [review]
There are some questions

File: dojo/tools/checkmarx/parser.py (hunk 2 above)
Comment (damiencarol) [reply dojo/tools/checkmarx/parser.py:78]: Yes. but better than what we have for now: it was only the language so the graphs were bugged.
My opinion is we don't have enough data so we should completely get ride of this.

File: dojo/tools/checkmarx/parser.py (hunk 3 above)
Comment (damiencarol) [reply dojo/tools/checkmarx/parser.py:109]: the format is not dependent of the OS. the data in the report are always like this: `WebGoat/webgoat-lessons/missing-function-ac/src/main/java/org/owasp/webgoat/plugin/Users.java`.
This version is better as it doesn't fail with empty strings.

File: dojo/tools/checkmarx/parser.py (hunk 1 above)
Comment (damiencarol) [reply dojo/tools/checkmarx/parser.py:47]: I fixed 3 of 4. I can complete this.

Comment (damiencarol) [reply dojo/tools/checkmarx/parser.py]: pushed

File: dojo/tools/checkmarx/parser.py (hunk 2 above)
Comment (StefanFl) [reply dojo/tools/checkmarx/parser.py:73]: I wouldn't miss it.

File: dojo/tools/checkmarx/parser.py (hunk 3 above)
Comment (StefanFl) [reply dojo/tools/checkmarx/parser.py:105]: ok

**Review by devGregA — APPROVED** [review]

File: dojo/tools/checkmarx/parser.py (hunk 2 above)
Comment (damiencarol) [reply dojo/tools/checkmarx/parser.py:73]: Let's do that in another PR. for now it keeps things as before, I will ask users and remove them if not required anymore

**Review by StefanFl — APPROVED** [review]

**Pull request merged**